olwebsite-app/src/contexts/UserContext.jsx

```diff
@@ -0,0 +1,45 @@
+import  React, { useContext, useEffect, useState } from 'react';
+
+const UserContext = React.createContext([undefined, undefined]);
+
+function UserProvider ({ children }) {
+    const [user, setUser] = useState(null);
+
+    useEffect(() => {
+        //TODO Jacob J: Don't use localStorage, very unsecure
+        const sessionUser = JSON.parse(window.localStorage.getItem('user'));
+        if (sessionUser && !user) {
+            setUser(sessionUser);
+        }
+    });
+
+    useEffect(() => {
```

Comment:
Is there a reason why you didn't combine the useEffect's?

Reply:
Yep. The second useEffect is only triggered when the user state changes, which could be the result of any situation where setUser is called and not necessarily just from that first useEffect's setUser. Where as the first one runs on every render (but has a check to ensure the user state isn't set every time).